zink/codegen/src/lib.rs

```diff
@@ -75,3 +75,41 @@ pub fn external(_args: TokenStream, input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as ItemFn);
     selector::external(input)
 }
+
+/// Bounds for Arithmetic Primitives
+trait SafeArithmetic: Sized {
+    fn safe_add(self, rhs: Self) -> Self;
+    fn safe_sub(self, rhs: Self) -> Self;
+    fn safe_mul(self, rhs: Self) -> Self;
+    fn safe_div(self, rhs: Self); 
+}
+
+macro_rules! impl_safe_arithmetic {
+    ($t:ty) => {
+        impl SafeArithmetic for $t {
+            fn safe_add(self, rhs: Self) -> Self {
+                self.checked_add(rhs)
```

Comment:
we can not use the `checked_add` from rust library because it will generate verbose bytecode in the WASM IR, instead, we need to append EVM assembly opcodes to operations, for example

```rust
fn safe_add(self, rhs: Self) -> Self {
   // the return value r is now on stack
   let r = self.add(rhs);
   if r > Self::MAX {
     zink::revert!("...");
   }
   r
}
```

Same for all of the other operations

Reply:
having some imorts issue, where should i place the logic

Reply:
replace them into your current functions of the SafeArithmetic trait, could please elaborate which kind of import issues you are facing?